features/configuration/fail_if_no_examples.feature

```diff
@@ -0,0 +1,106 @@
+   when there are 0 examples.
+
+  Scenario: A failing spec example and `fail_if_no_examples` set to true
+    Given a file named "spec/spec_helper.rb" with:
+      """ruby
+      RSpec.configure { |c| c.fail_if_no_examples = true }
+      """
+    Given a file named "spec/example_spec.rb" with:
+      """ruby
+      require 'spec_helper'
+
+      RSpec.describe "something" do
+        it "fails" do
+          fail
+        end
+      end
+      """
+    When I run `rspec spec/example_spec.rb`
+    Then the exit status should be 1
+
+  Scenario: A failing spec example and `fail_if_no_examples` set to false
+    Given a file named "spec/spec_helper.rb" with:
+      """ruby
+      RSpec.configure { |c| c.fail_if_no_examples = false }
+      """
+    Given a file named "spec/example_spec.rb" with:
+      """ruby
+      require 'spec_helper'
+
+      RSpec.describe "something" do
+        it "fails" do
+          fail
+        end
+      end
+      """
+    When I run `rspec spec/example_spec.rb`
+    Then the exit status should be 1
+
+  Scenario: A passing spec example and `fail_if_no_examples` set to true
+    Given a file named "spec/spec_helper.rb" with:
+      """ruby
+      RSpec.configure { |c| c.fail_if_no_examples = true }
+      """
+    Given a file named "spec/example_spec.rb" with:
+      """ruby
+      require 'spec_helper'
+
+      RSpec.describe "something" do
+        it "succeeds" do
+          true
+        end
+      end
+      """
+    When I run `rspec spec/example_spec.rb`
+    Then the exit status should be 0
+
+  Scenario: A passing spec example and `fail_if_no_examples` set to false
+    Given a file named "spec/spec_helper.rb" with:
+      """ruby
+      RSpec.configure { |c| c.fail_if_no_examples = false }
+      """
+    Given a file named "spec/example_spec.rb" with:
+      """ruby
+      require 'spec_helper'
+
+      RSpec.describe "something" do
+        it "succeeds" do
+          true
+        end
+      end
+      """
+    When I run `rspec spec/example_spec.rb`
+    Then the exit status should be 0
+
+  Scenario: No examples and `fail_if_no_examples` set to true
+    Given a file named "spec/spec_helper.rb" with:
+      """ruby
+      RSpec.configure { |c| c.fail_if_no_examples = true }
+      """
+    Given a file named "spec/example_spec.rb" with:
+      """ruby
+      require 'spec_helper'
+
+      RSpec.describe "something" do
+      end
+      """
+    When I run `rspec spec/example_spec.rb`
+    Then the exit status should be 1
+
+  Scenario: No examples and `fail_if_no_examples` set to false
+   Given a file named "spec/spec_helper.rb" with:
+     """ruby
+     RSpec.configure { |c| c.fail_if_no_examples = false }
+     """
+   Given a file named "spec/example_spec.rb" with:
+     """ruby
+     require 'spec_helper'
+
+     RSpec.describe "something" do
+     end
+     """
+   When I run `rspec spec/example_spec.rb`
+   Then the exit status should be 0
```

Comment:
Initially I wanted to add a test similar to https://github.com/rspec/rspec-core/blob/master/spec/rspec/core/runner_spec.rb#L431 , but I couldn't make it work: I got `LoadError` when adding:
```ruby
context "when fail_if_no_examples option is set" do
  include_context "spec files"

  it "returns 0 if examples pass and fail_if_no_examples is true" do
    runner = build_runner failing_spec_filename, "fail_if_no_examples", "true"
    expect(runner.run(err, out)).to eq 0
  end
end
```
It seemed appropriate place for such test, but since I cannot make it work and you suggest writing `spec/integration`, I'll provide such tests.